salt/config/__init__.py

```diff
@@ -2560,8 +2560,6 @@ def is_profile_configured(opts, provider, profile_name):
 
     if driver not in non_image_drivers:
         required_keys.append('image')
-    elif driver == 'vmware':
-        required_keys.append('clonefrom')
```

Comment:
Could you please elaborate a little on what the check is there for? Seems to me like the VMware driver really doesn't match the cloud provider mould and if this commit makes the clonefrom key optional it should be removed.

I respectfully disagree on renaming guest_id to image since an image in another cloud provider is a completely different thing from guest_id in VMware. However since you have waay more experince with all of this than I do I will rename it.